File: emotiv-sdk/examples_extra_prime/C++/ActivateLicense/ActivateLicense.cpp
/****************************************************************************
**
** Copyright 2016 by Emotiv. All rights reserved
** Example - ActivateLicense
** How to activate a license key and Get infomation of that license 
**
****************************************************************************/

#ifdef _WIN32
    #include <windows.h>
    #include <conio.h>
#endif

#if __linux__ || __APPLE__
#include <unistd.h>
#include <termios.h>
int _getch(void);
#endif

#include <iostream>
#include <sstream>
#include <time.h>
#include <stdio.h>
#include <ctime>

#include "Iedk.h"
#include "IedkErrorCode.h"

std::string convertEpochToTime(time_t epochTime, std::string format = "%Y-%m-%d %H:%M:%S");
std::string convertEpochToTime(time_t epochTime, std::string format)
{
    if (format == "")
        format = "%Y-%m-%d %H:%M:%S";


    char timestamp[64] = { 0 };
    strftime(timestamp, sizeof(timestamp), format.c_str(), localtime(&epochTime));
    return timestamp;
}

std::string intToHex(int x)
{
    std::stringstream stream;
    stream << std::hex << x;
    std::string result(stream.str());
    result = "0x" + result;

    return result;
}

void printLicenseInformation(IEE_LicenseInfos_t& licenseInfos)
{
    int licenseType = 0;

    std::cout << std::endl;
    std::cout << "Date From  : " << convertEpochToTime(licenseInfos.date_from) << std::endl;
    std::cout << "Date To    : " << convertEpochToTime(licenseInfos.date_to) << std::endl;
    std::cout << std::endl;

    std::cout << "Seat number: " << licenseInfos.seat_count << std::endl;
    std::cout << std::endl;

    std::cout << "Total Quota: " << licenseInfos.quota << std::endl;
    std::cout << "Total quota used    : " << licenseInfos.usedQuota << std::endl;
    std::cout << std::endl;

    std::cout << "Quota limit in day  : " << licenseInfos.quotaDayLimit << std::endl;
    std::cout << "Quota used in day   : " << licenseInfos.usedQuotaDay << std::endl;
    std::cout << std::endl;

    std::cout << "Quota limit in month: " << licenseInfos.quotaMonthLimit << std::endl;
    std::cout << "Quota used in month : " << licenseInfos.usedQuotaMonth << std::endl;
    std::cout << std::endl;

    switch (licenseInfos.scopes)
    {
    case IEE_EEG:
        licenseType = IEE_LicenseType_t::IEE_EEG;

        std::cout << "License type : " << "EEG" << std::endl;
        std::cout << std::endl;
        break;
    case IEE_EEG_PM:
        licenseType = IEE_LicenseType_t::IEE_EEG_PM;

        std::cout << "License type : " << "EEG + PM" << std::endl;
        std::cout << std::endl;
        break;
    case IEE_PM:
        licenseType = IEE_LicenseType_t::IEE_PM;
        std::cout << "License type : " << "PM" << std::endl;
        std::cout << std::endl;
        break;
    default:
        std::cout << "License type : " << "No type" << std::endl;
        std::cout << std::endl;
        break;
    }
}

std::string const LICENSE_KEY = "";

int main(int argc, char** argv)
{

    int result = EDK_OK;

    // Need only 1 time to call with the internet connection
    result = IEE_ActivateLicense(LICENSE_KEY.c_str());    

    switch (result)
    {
        case EDK_LICENSE_NOT_FOUND:
        case EDK_LICENSE_ERROR:
        case EDK_LICENSE_EXPIRED:
        case EDK_LICENSE_REGISTERED:
        case EDK_LICENSE_DEVICE_LIMITED:
        case EDK_UNKNOWN_ERROR:
            break;

        default:
            break;
    }

    if (!(result == EDK_OK || result == EDK_LICENSE_REGISTERED))
        return result;
    
    IEE_LicenseInfos_t licenseInfos;

    // We can call this API any time to check current License information
    result = IEE_LicenseInformation(&licenseInfos);

    std::cout << std::endl << "IEE_LicenseInfomation result = " << intToHex(result) << std::endl;

    switch (result)
    {
        case EDK_OVER_QUOTA_IN_DAY:
        case EDK_OVER_QUOTA_IN_MONTH:
        case EDK_LICENSE_EXPIRED:
        case EDK_OVER_QUOTA:
        case EDK_ACCESS_DENIED:
        case EDK_LICENSE_ERROR:
            break;
        case EDK_NO_ACTIVE_LICENSE:
            break;
        case EDK_OK:
            break;
        default:
            break;
    }

    printLicenseInformation(licenseInfos);

    _getch();

}
#ifdef __APPLE__
int _getch(void)
{
    int r;
    unsigned char c;
    if ((r = read(0, &c, sizeof(c))) < 0)
    {
        return r;
    }
    else
    {
        return c;
    }
}
#endif

File: emotiv-sdk/examples_extra_prime/C#/ActivateLicense/ActivateLicense.cs
/****************************************************************************
**
** Copyright 2015 by Emotiv. All rights reserved
** Example - ActivateLicense
** How to activate a license key
 ***
****************************************************************************/

using System;
using System.Collections.Generic;
using Emotiv;
using System.IO;
using System.Threading;
using System.Reflection;

namespace ActivateLicense
{
    class ActivateLicense
    {
        static string licenseKey = "";           // Your License Key

        static string format = "dd/MM/yyyy";

        static void activateLicense()
        {
            int result = EdkDll.IEE_ActivateLicense(licenseKey);
            if (result == EdkDll.EDK_OK || result == EdkDll.EDK_LICENSE_REGISTERED)
            {
                Console.WriteLine("License activated.");
            }
            else Console.WriteLine("License Error. " + result);
        }

        static public DateTime FromUnixTime(uint unixTime)
        {
            var epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            return epoch.AddSeconds(unixTime);
        }

        static public long ToUnixTime(DateTime date)
        {
            var epoch = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            return Convert.ToInt64((date - epoch).TotalSeconds);
        }

        static void licenseInformation()
        {
            EdkDll.IEE_LicenseInfos_t licenseInfos = new EdkDll.IEE_LicenseInfos_t();
            int result = EdkDll.IEE_LicenseInformation(ref licenseInfos);

            Console.WriteLine();
            Console.WriteLine("Date From  : " + FromUnixTime(licenseInfos.date_from).ToString(format));
            Console.WriteLine("Date To    : " + FromUnixTime(licenseInfos.date_to).ToString(format));
            Console.WriteLine();

            Console.WriteLine("Seat number: " + licenseInfos.seat_count);
            Console.WriteLine();

            Console.WriteLine("Total Quota: " + licenseInfos.quota);
            Console.WriteLine("Total quota used    : " + licenseInfos.usedQuota);
            Console.WriteLine();

            Console.WriteLine("Quota limit in day  : " + licenseInfos.quotaDayLimit);
            Console.WriteLine("Quota used in day   : " + licenseInfos.usedQuotaDay);
            Console.WriteLine();

            Console.WriteLine("Quota limit in month: " + licenseInfos.quotaMonthLimit);
            Console.WriteLine("Quota used in month : " + licenseInfos.usedQuotaMonth);
            Console.WriteLine();

            switch ((int)licenseInfos.scopes)
            {
                case (int)EdkDll.IEE_LicenseType_t.IEE_EEG:

                    Console.WriteLine("License type : EEG");
                    Console.WriteLine();
                    break;
                case (int)EdkDll.IEE_LicenseType_t.IEE_EEG_PM:

                    Console.WriteLine("License type : EEG + PM");
                    Console.WriteLine();
                    break;
                case (int)EdkDll.IEE_LicenseType_t.IEE_PM:
                    Console.WriteLine("License type : PM");
                    Console.WriteLine();
                    break;
                default:
                    Console.WriteLine("License type : No type");
                    Console.WriteLine();
                    break;
            }
        }

        static void Main(string[] args)
        {
            Console.WriteLine("===========================================");
            Console.WriteLine("The example to activate a license key.");
            Console.WriteLine("===========================================");

            //Need only 1 time to call with the internet connection
            activateLicense();

            //We can call this API any time to check current License information
            licenseInformation();

            Thread.Sleep(5000);
        }
    }
}
